Docs: Initial docs cleanup

mtodor opened the pull request:
## Description

This is a product of the initial pass over the current documentation with @SimonBaeumer.

We have made the following changes:
- several not relevant or not useful documents are deleted
- mermaid files are deleted
- outdated documents are moved to the legacy directory to revisit them and update
- docs are organized into `development` guides and other globally relevant docs, like `auth`, `observability`, etc.
- package specific docs are moved to related packages

## Additional cleanup

We need help regarding several files to get feedback from other team members if we can delete them or we should keep them.

Please provide feedback on the listed files and directories. Can we delete them, or should we move them to legacy for revisiting?
| File / Dir | Team member(s) | Should be deleted? | Comment |
|---|---|---|---|
| `docs/observability` | @kurlov @stehessel | [➖] | Let's keep it for sneak peek. It will be used to generate a new readme uner `dashboards` folder |
| `docs/slos` | @kurlov @stehessel | [➖] | Let's keep it and drop it once we settle our SLOs with appSRE |
| `docs/legacy/deploying-fleet-manager-to-openshift.md` | @kurlov | [✔️] | |
| `templates/README.md` | @kurlov | [✔️] | |
| `docs/legacy/data-plane-osd-cluster-options.md` | @ebensh | [?] | The heading "Using an existing OSD cluster with manual scaling enabled" was useful for defining the data plane / knowing what the configuration file is. Maybe integrate into docs elsewhere, then this can be deleted |
| `docs/legacy/getting-credentials-and-accounts.md` | @ebensh | [✔️] | I believe this is documented well in https://docs.engineering.redhat.com/display/StackRox/Creating+RH+Org%2C+OCM+Quota%2C+AWS+Acct%2C+OSD+Cluster |
| `docs/legacy/onboarding-with-service-delivery.md` | @ebensh @kurlov  | [➖] | IMHO this is up to Aleks who configured the Envoy - is it useful to have the Envoy information here or do we have it elsewhere e.g. in App-Interface itself |
| `dev/env/README.md` | @mtesseract | [ ] | This doc looks like good candidate for development docs. Should we move this to `docs/development` |
| `dev/env/central-to-ready/TESTING-HOWTO.md` | @mtesseract | [ ] | This doc looks like good candidate for development docs. Should we move this to `docs/development` |
| `pkg/services/queryparser/complex-string-parsing-and-validation.md` | @SimonBaeumer | [ ] | Can you please check with Kafka team if we still need this? |

Please comment in PR, and I'll delete files in this PR or move files to legacy if they should be revisited.

## Checklist (Definition of Done)
- ~~[ ] Unit and integration tests added~~
- [x] Documentation added if necessary
- ~~[ ] CI and all relevant tests are passing~~

## Test manual

This PR includes deletion and small restructuring of documentation. There is nothing to test.

kurlov commented on knowledge.md:
```diff
@@ -1,3 +0,0 @@
-# Knowledge
```
Ernest Hemingway once won a bet by crafting a six-word short story that can make people cry. Here it is. "Removing 3-lines `knowledge.md`. Not used.".

kurlov reviewed (approved):
Looking great! 🎉

kurlov commented on knowledge.md:
```diff
@@ -1,3 +0,0 @@
-# Knowledge
-
-This document is used to collect useful code snippets, pointers, very short scripts, etc. related to fleet management and service delivery.
```
Ernest Hemingway once won a bet by crafting a six-word short story that can make people cry. Here it is. "Removing 3-lines knowledge.md. Not used.".

mtodor commented:
All comments addressed. Thank you for the help!

@mtesseract - I have renamed files and moved to `docs/development`. There have been 2 small changes within the files because we had statements like `this directory contains`.
@kurlov - I have moved `slos` and `observability` to `legacy`. When we revisit these files, we can pull them out of legacy.
@SimonBaeumer - I have seen the comment in Zulip and changed the file accordingly.
